### PR TITLE
fix tcsetattr: Interrupted system call

### LINKED ISSUE #2170: script that calls ssh gets interrupted
```
laerus@xonsh$ cat autossh 
ssh laerus@192.168.1.10
laerus@xonsh$ ./autossh 
^Z <--- didn't press anything here, this comes from xonsh or ssh
```

tried again with a non-lan remote
```
laerus@xonsh$ ./autossh
^Z
tcsetattr: Interrupted system call
```

### PR DESCRIPTION
**Pull request by mitnk**
try fixes #2170

### CODE REVIEW

File: xonsh/built_ins.py [hunk 1]
```diff
@@ -535,6 +535,9 @@ def prep_preexec_fn(self, kwargs, pipeline_group=None):
         """Prepares the 'preexec_fn' keyword argument"""
         if not (ON_POSIX and self.cls is subprocess.Popen):
             return
+        env = builtins.__xonsh_env__
+        if not env['XONSH_INTERACTIVE']:
```
Comment (scopatz) [inline xonsh/built_ins.py]: This might need to be env.get('XONSH_INTERACTIVE')

Comment (astronouth7303) [inline xonsh/built_ins.py]: `'XONSH_INTERACTIVE' in env`?

Comment (mitnk) [inline xonsh/built_ins.py]: I think this is safe. this env is always set during the start up phase, and here is a run-sub-proc phase. But I could update a bit :)

Comment (mitnk) [inline xonsh/built_ins.py]: Ok, I see the points. CI was failing because of this. In tests we may only mock part of the ENVs.

**Comment by scopatz** [comment]
This seems like a reasonable change. thanks @mitnk!

**Pull request merged**